### PR TITLE
Setup: Allow to specify optional deps using pip

### PR DESCRIPTION
**Pull request by nijel**
The full version can be installed by:

    pip install translate-toolkit[all]

See #4004

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -181,6 +181,30 @@
...
+                extras_require[section] = [dependency]
+            else:
+                extras_require[section].append(dependency)
+            extras_require["all"].append(dependency)
+            if recommended:
+                extras_require["recommended"].append(dependency)
+
+
+extras_require = {"recommended": [], "all": []}
+update_reqires("requirements/optional.txt", extras_require, False)
```
Comment (unho) [inline setup.py:205]: reqires?

Comment (nijel) [reply setup.py:205]: It's called `extras_require` in the setup args, so I kept it same here (see below in the patch).

Comment (unho) [reply setup.py:205]: `reqires` != `requires`

Comment (nijel) [reply setup.py]: Sorry, I didn't notice that.

**Comment by unho** [comment]
lgtm

**Pull request merged**